core/src/test/java/org/opensearch/sql/expression/datetime/UnixTimeStampTest.java

```diff
@@ -0,0 +1,235 @@
+    assertEquals(System.currentTimeMillis() / 1000L, unixTimeStamp());
+    assertEquals(System.currentTimeMillis() / 1000L, eval(unixTimeStampExpr()).longValue());
+  }
+
+  private static Stream<Arguments> getDateSamples() {
+    return Stream.of(
+        Arguments.of(LocalDate.of(1984, 1, 1)),
+        Arguments.of(LocalDate.of(2000, 2, 29)),
+        Arguments.of(LocalDate.of(1999, 12, 31)),
+        Arguments.of(LocalDate.of(2004, 2, 29)),
+        Arguments.of(LocalDate.of(2100, 2, 28)),
+        Arguments.of(LocalDate.of(2012, 2, 21))
+    );
+  }
+
+  /**
+   * Check processing valid values of type LocalDate.
+   * @param value a value
+   */
+  @ParameterizedTest
+  @MethodSource("getDateSamples")
+  public void checkOfDate(LocalDate value) {
+    assertEquals(value.getLong(ChronoField.EPOCH_DAY) * 24 * 3600, unixTimeStampOf(value));
+    assertEquals(value.getLong(ChronoField.EPOCH_DAY) * 24 * 3600,
+        eval(unixTimeStampOf(DSL.literal(new ExprDateValue(value)))).longValue());
+  }
+
+  private static Stream<Arguments> getDateTimeSamples() {
+    return Stream.of(
+        Arguments.of(LocalDateTime.of(1984, 1, 1, 1, 1)),
+        Arguments.of(LocalDateTime.of(2000, 2, 29, 22, 54)),
+        Arguments.of(LocalDateTime.of(1999, 12, 31, 23, 59)),
+        Arguments.of(LocalDateTime.of(2004, 2, 29, 7, 40)),
+        Arguments.of(LocalDateTime.of(2100, 2, 28, 13, 14)),
+        Arguments.of(LocalDateTime.of(2012, 2, 21, 0, 0))
+    );
+  }
+
+  /**
+   * Check processing valid values of type LocalDateTime.
+   * @param value a value
+   */
+  @ParameterizedTest
+  @MethodSource("getDateTimeSamples")
+  public void checkOfDateTime(LocalDateTime value) {
+    assertEquals(value.toEpochSecond(ZoneOffset.UTC), unixTimeStampOf(value));
+    assertEquals(value.toEpochSecond(ZoneOffset.UTC),
+        eval(unixTimeStampOf(DSL.literal(new ExprDatetimeValue(value)))).longValue());
+  }
+
+  private static Stream<Arguments> getInstantSamples() {
+    return getDateTimeSamples()
+        .map(v -> Arguments.of(((LocalDateTime)v.get()[0]).toInstant(ZoneOffset.UTC)));
+  }
+
+  /**
+   * Check processing valid values of type Instant.
+   * @param value a value
+   */
+  @ParameterizedTest
+  @MethodSource("getInstantSamples")
+  public void checkOfInstant(Instant value) {
+    assertEquals(value.getEpochSecond(), unixTimeStampOf(value));
+    assertEquals(value.getEpochSecond(),
+        eval(unixTimeStampOf(DSL.literal(new ExprTimestampValue(value)))).longValue());
+  }
+
+  // formats: YYMMDD, YYMMDDhhmmss[.uuuuuu], YYYYMMDD, or YYYYMMDDhhmmss[.uuuuuu]
+  // use BigDecimal, because double can't fit such big values
+  private static Stream<Arguments> getDoubleSamples() {
+    return Stream.of(
+        Arguments.of(840101d, LocalDateTime.of(1984, 1, 1, 0, 0, 0)),
+        Arguments.of(840101112233d, LocalDateTime.of(1984, 1, 1, 11,22,33)),
+        Arguments.of(840101112233.123456, LocalDateTime.of(1984, 1, 1, 11, 22, 33, 123456000)),
```

Comment:
do the `argument.of`'s need a `d` at the end too?

Reply:
`d` for Double, otherwise it is interpreted as int/long and test crashes, because the test method expect `double` as the first arg.